include/seqan3/range/view/async_buffer.hpp

```diff
@@ -0,0 +1,538 @@
+            "value of its reference type.");
+
+    //!\brief The iterator type for the underlying range.
+    using urng_iterator_type = std::ranges::iterator_t<urng_t>;
+
+    //!\brief The underlying range.
+    urng_t urange;
+
+    //!\brief The buffer queue.
+    contrib::fixed_buffer_queue<value_type_t<urng_t>> buffer;
+
+    //!\brief Iterates over the underlying range and pushes into the buffer until at end or stopped.
+    void produce()
+    {
+        auto it = std::ranges::begin(urange);
+        auto e  = std::ranges::end(urange);
+
+        detail::spin_delay delay{};
+
+        while ((it != e) && (!stop_producer.load()))// TODO likely
+        {
+            auto tmp = std::move(*it);
+
+            while (!stop_producer.load())
+            {
+                auto status = buffer.try_push(std::move(tmp));
+                if (status == contrib::queue_op_status::full) // queue was full
+                    delay.wait();
+                else                                          // push successful
+                    break;
+            }
+
+            ++it; // move to next item in source
+        }
+
+        buffer.close();
+    }
+
+    //!\brief Thread that runs #produce in the background.
+    std::thread producer;
+
+    //!\brief Abort thread before it reaches the end.
+    std::atomic<bool> stop_producer{false};
+
+    //!\brief The iterator of the view.
+    class async_buffer_iterator
+    {
+        //!\brief The sentinel type to compare to.
+        using sentinel_type      = std::ranges::default_sentinel_t;
+
+        //!\brief The pointer to the associated view.
+        contrib::fixed_buffer_queue<value_type_t<urng_t>> * buffer_ptr = nullptr;
+
+        //!\brief The cached value this iterator holds.
+        value_type_t<urng_t> cached_value;
+
+        //!\brief Whether this iterator is at end (the buffer is empty and closed).
+        bool at_end = false;
+
+    #ifndef NDEBUG
+        //!\brief Whether the iterator currently holds a valid value in #cached_value.
+        bool has_value = false;
+    #endif
+
+    public:
+
+        /*!\name Associated types
+        * \{
+        */
+        //!\brief Difference type.
+        using difference_type   = difference_type_t<urng_iterator_type>;
+        //!\brief Value type.
+        using value_type        = value_type_t<urng_iterator_type>;
+        //!\brief Pointer type.
+        using pointer           = value_type*;
+        //!\brief Reference type.
+        using reference         = value_type;
+        //!\brief Iterator category.
+        using iterator_category = std::input_iterator_tag;
+        //!\}
+
+        /*!\name Construction, destruction and assignment
+        * \{
+        */
+        async_buffer_iterator()                                              noexcept = default; //!< Defaulted.
+        //TODO: delete:
+        async_buffer_iterator(async_buffer_iterator const & rhs)             noexcept = default; //!< Deleted.
```

Comment:
Mhm shouldn't it be `//!< Defaulted.` then until it is actually deleted?
Same with copy assignment.

Reply:
Hm, yeah, I don't care!